# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowcreator/nsIWindowProvider.idl
# Commit: 765d3364cca0
# Full Hash: 765d3364cca0a67c614cb4fb5dff413a8c29d834
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-07 03:53:09
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 7.2: Create and use nsOpenWindowInfo types in nsWindowWatcher logic, r=kmag
#   
#   This patch builds on top of part 7.1 by creating this object within
#   nsWindowWatcher and ContentParent to carry the relevant information through
#   provider interfaces when creating new content windows. The nsOpenWindowInfo
# ==============================================================================

diff -r 8a13355b4ac4 -r 765d3364cca0 toolkit/components/windowcreator/nsIWindowProvider.idl
--- a/toolkit/components/windowcreator/nsIWindowProvider.idl	Mon Apr 06 14:29:59 2020 +0000
+++ b/toolkit/components/windowcreator/nsIWindowProvider.idl	Mon Apr 06 14:30:02 2020 +0000
@@ -20,6 +20,7 @@
 webidl BrowsingContext;
 interface mozIDOMWindowProxy;
 interface nsIURI;
+interface nsIOpenWindowInfo;
 native nsDocShellLoadStatePtr(nsDocShellLoadState*);
 
 /**
@@ -39,11 +40,8 @@
    * those is the caller's responsibility.  The provider can always return null
    * to have the caller create a brand-new window.
    *
-   * @param aParent Must not be null.  This is the window that the caller wants
-   *        to use as the parent for the new window.  Generally,
-   *        nsIWindowProvider implementors can expect to be somehow related to
-   *        aParent; the relationship may depend on the nsIWindowProvider
-   *        implementation.
+   * @param aOpenWindowInfo  Must not be null.  This is the information the
+   *                         caller wants to be used to construct the new window.
    *
    * @param aChromeFlags The chrome flags the caller will use to create a new
    *        window if this provider returns null.  See nsIWebBrowserChrome for
@@ -83,7 +81,7 @@
    *        false if the window being returned existed before the
    *        provideWindow() call.  The value of this out parameter is
    *        meaningless if provideWindow() returns null.
-
+   *
    * @return A window the caller should use or null if the caller should just
    *         create a new window.  The returned window may be newly opened by
    *         the nsIWindowProvider implementation or may be a window that
@@ -95,7 +93,7 @@
    * @see nsIWindowWatcher for more information on aFeatures.
    * @see nsIWebBrowserChrome for more information on aChromeFlags.
    */
-  BrowsingContext provideWindow(in mozIDOMWindowProxy aParent,
+  BrowsingContext provideWindow(in nsIOpenWindowInfo aOpenWindowInfo,
                                 in unsigned long aChromeFlags,
                                 in boolean aCalledFromJS,
                                 in boolean aWidthSpecified,